src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -623,6 +742,24 @@ private static void WriteData(Stream archiveStream, Stream dataStream, long actu
             }
         }
 
+        // Calculates the padding for the current entry and asynchronously writes it after the data.
+        private static ValueTask WritePaddingAsync(Stream archiveStream, long actualLength, CancellationToken cancellationToken)
+        {
+            int paddingAfterData = TarHelpers.CalculatePadding(actualLength);
+            if (paddingAfterData != 0)
+            {
+                Debug.Assert(paddingAfterData <= TarHelpers.RecordSize);
+
+                byte[] b = new byte[TarHelpers.RecordSize];
+                Memory<byte> padding = b.AsMemory();
+                padding = padding.Slice(0, paddingAfterData);
+
+                return archiveStream.WriteAsync(padding, cancellationToken);
```

Comment:
Why allocate an array of size X (`TarHelpers.RecordSize`) just to slice it to size Y (`paddingAfterData`)?

```suggestion
                byte[] zeros = new byte[paddingAfterData];
                
                return archiveStream.WriteAsync(zeros, cancellationToken);
```

I assume that `paddingAfterData` is small and renting an array would not be beneficial.

Reply:
We're not supposed to use variable allocation lengths when creating stackallocs.

Although the CalculatePadding method will return a number under 512, if that method ever changes its logic unexpectedly and stops having that upper limit, we could get in trouble.

There is a debug assert that confirms that paddingAfterData is smaller than RecordSize, but still.

BTW this was actually very nicely explained by our very own @vcsjones in his blog post https://vcsjones.dev/stackalloc/

Reply:
> when creating stackallocs

That is correct, but this method is async and you are not using `stackalloc` but `new`.

Reply:
Ah good point. I was looking at the sync one.

Reply:
I assume the same rule of avoiding using a variable length does not apply when creating a byte array?